forc-plugins/forc-publish/src/main.rs

```diff
@@ -0,0 +1,52 @@
+use clap::{crate_version, Parser};
+use forc_publish::credentials::get_auth_token;
+use forc_publish::error::Result;
+use forc_publish::forc_pub_client::ForcPubClient;
+use forc_publish::tarball::create_tarball_from_current_dir;
+use forc_tracing::{
+    init_tracing_subscriber, println_action_green, println_error, TracingSubscriberOptions,
+};
+use tempfile::tempdir;
+
+// For local development, change to: http://localhost:8080
+const FORC_PUB_URL: &str = "https://forc-pub-dev.swayswap.io";
+
+#[derive(Parser, Debug)]
+#[clap(name = "forc-publish", version)]
+/// Forc plugin for uploading packages to the registry.
+pub struct Opt {
+    /// Token to use when uploading
+    #[clap(long)]
+    pub token: Option<String>,
+}
```

Comment:
Maybe we should make the url a CLI param - which defaults to `FORC_PUB_URL`; but can be overriden (to localhost for example).